indexer/indexer.go

```diff
@@ -72,22 +73,30 @@ func NewIndexer(ctx *cli.Context) (*Indexer, error) {
 		return nil, err
 	}
 
-	// L1 Processor
+	// L1 Processor (hardhat devnet contracts). Make this configurable
+	l1Contracts := processor.L1Contracts{
+		OptimismPortal:         common.HexToAddress("0x6900000000000000000000000000000000000000"),
+		L2OutputOracle:         common.HexToAddress("0x6900000000000000000000000000000000000001"),
+		L1CrossDomainMessenger: common.HexToAddress("0x6900000000000000000000000000000000000002"),
+		L1StandardBridge:       common.HexToAddress("0x6900000000000000000000000000000000000003"),
+		L1ERC721Bridge:         common.HexToAddress("0x6900000000000000000000000000000000000004"),
+	}
```

Comment:
Specifying contract addresses like this is a mistake Optimism consistently makes imo.   doesn't have to be this pr but we really need to make it a point to try to use deployment artifacts as a single source of truth

Reply:
might go to client office hours to ask about this.  Easiest solution imo is to have a go package which is generated from the artifacts

Reply:
There might actually already exist a solution like getContractArftifacts in js that I'm just not aware of as well

Reply:
also one more comment about this.   These are actually coming from the config.toml file.   It has a section in it called [chain] which would be where these are configured.    I'm including a property though called `preset` that allows you to just use known chain presets for known chains like OP and base without having to specify everything though

Reply:
I'm all in for having presets that can be selected. Kinda left that open ended on the configuration side of things

The reason its important for the processor to not bias towards preset it so that it can work with any OP chain

Reply:
tbh that part is intuitive it's more like "wtf is this language feature" that's confusing me.  Seeing here it's kinda an extra property Tag ont he thing though which makes it look less wierd/magical https://medium.com/golangspec/tags-in-golang-3e5db0b8ef3e

Reply:
presets aren't doing anything magical.   A preset is just something that specifies all the contract addresses for you without having to be verbose about it.   None of the application will have any idea that presets exist

Reply:
What I was getting at when mentioning the config and presets is that this code you wrote here should not be here.  It has tob e passed in from teh cli app as a param

Reply:
essentially nothing about the app other than the config should be aware of contract addresses and other configurable things

Reply:
gotcha i see what you mean. On the same page that the config the sole entry point for how these addresses are generated.

We definitely don't have a good approach to this now. I can also smell future complexity once we go through additional contract upgrades where the ABI has to be versioned

Reply:
Yup that's one reason I really want the build pipeline between contract artifacts and services like this to be as tight as possible.   If it's tight, we avoid versioning issues because the contracts upgrade and the indexer upgrade are forced to stay in sync

Reply:
>  If it's tight, we avoid versioning issues because the contracts upgrade and the indexer upgrade are forced to stay in sync

I don't think this enforcement would be desired for the indexer. For example, if someone is resyncing the indexer from genesis, using the latest version isn't actually what you want. The lifecycle of an op-chain will consist of many versions/upgrades

Reply:
resolving this to merge PR but good conversation to follow up on